backend/src/api/public/media/media.controller.ts

```diff
@@ -101,6 +104,17 @@ export class MediaController {
     return await this.mediaService.toMediaUploadDto(upload);
   }
 
+  @Get(':filename')
+  @OpenApi(404, 500)
+  async getMedia(
+    @Param('filename') filename: string,
+    @Res() response: Response,
+  ): Promise<void> {
+    const mediaUpload = await this.mediaService.findUploadByFilename(filename);
+    const targetUrl = mediaUpload.fileUrl;
+    response.redirect(targetUrl);
```

Comment:
Do we need to implement  some kind of check if the user has access to the media? Otherwise a user with an API Token could theoretically try to find files with predictable names (e.g Screenshot_$date.png).

Reply:
As the filename get's randomized, this is not really a concern…